src/lin.py

```diff
@@ -16,7 +16,7 @@ def main():
     fastq = p.parseFastq(args.reads)
     for (fastaName, fastaSeq) in fasta:
         for (name,seq) in fastq:
-            for i in kmp(fastaSeq, seq):
+            for i in bmh(fastaSeq, seq):
```

Comment:
Great you have `bmh`, but keep in mind that it doesn't run in $O(n+m)$ but $O(nm)$ worst-case.